src/System.Buffers.Primitives/System/Buffers/Pooling/ArrayMemoryPool.cs

```diff
@@ -10,12 +9,19 @@ namespace System.Buffers.Internal
 {
     internal sealed class ArrayMemoryPool<T> : MemoryPool<T>
     {
+        const int DefaultSize = 4096;
+
         readonly static ArrayMemoryPool<T> s_shared = new ArrayMemoryPool<T>();
 
         public static ArrayMemoryPool<T> Shared => s_shared;
 
-        public override OwnedMemory<T> Rent(int minimumBufferSize)
+        public override int MaxBufferSize => 1024 * 1024 * 1024;
+
+        public override OwnedMemory<T> Rent(int minimumBufferSize = DefaultSize)
         {
+            if (minimumBufferSize == 0) minimumBufferSize = 32;
+            else if (minimumBufferSize == AnySize) minimumBufferSize = DefaultSize;
```

Comment:
Why do we support both `AnySize` and `0`?

Reply:
What would you prefer to happen when 0 is passed?

Reply:
We need to decide what const are we using for `AnySize` and throw in all other non-valid cases.

Reply:
OK, I will change it to throw for zero